src/main/java/net/dv8tion/jda/api/entities/Invite.java

```diff
@@ -503,4 +589,70 @@ default String getUrl()
         GROUP,
         UNKNOWN
     }
+
+    /**
+     * Enum representing the target type of an invite.
+     *
+     * @see #getTargetType()
+     */
+    enum TargetType
+    {
+        /**
+         * The invite points to an users stream in a voice channel.
+         * The user to whose stream the invite goes can be get with {@link Invite#getTargetUser() Invite.getTargetUser} and is not {@code null}.
+         *
+         * @see Invite#getTargetUser()
+         */
+        STREAM(1),
+
+        /**
+         * The invite points to an application in a voice channel.
+         * The application to which the invite goes can be get with {@link Invite#getTargetApplication() Invite.getTargetApplication} and is not {@code null}.
+         *
+         * @see Invite#getTargetApplication()
+         */
+        EMBEDDED_APPLICATION(2),
+
+        /**
+         * Unknown invite target type.
+         * Either this invite does not have a target type or Discord implemented a new invite target type and JDA has not yet implemented support for it.
+         */
+        UNKNOWN(-1);
```

Comment:
Is there no type `0` to represent "No target type"?
It may be useful to represent the the invite has no target at all compared to "we don't know what the target is due to not implementing it"

Reply:
When no target type is set for the invite discord just leaves this property out.
We could use `0` as default value which represents `TargetType.NONE` or something like that.

Reply:
> We could use `0` as default value which represents `TargetType.NONE` or something like that.

Added that in the latest commit, would it better to use `null` in this place?